models/Version/Adapter/DelegateVersionStorageAdapter.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Version\Adapter;
+
+class DelegateVersionStorageAdapter implements VersionStorageAdapterInterface
+{
+    private array $adapters = [];
+    public function __construct(protected int $byte_threshold,
```

Comment:
```suggestion
    public function __construct(protected int $byteThreshold,
```
Variable names should always be camelCase, snake_case is only used by Symfony configs.